src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BuldozerProduser().getMachines());
+        }
+        if (type == Excavator.class) {
+            return new ArrayList<>(new ExcavatorProduser().getMachines());
+        }
+        if (type == Truck.class) {
+            return new ArrayList<>(new TruckProduser().getMachines());
```

Comment:
Do not create Produsers each time you call getAll(). think, how you can create All producers only one time for this class.